### PR TITLE
Demystifying Terraform Modules

### PR DESCRIPTION
**Pull request by kemanedonfack**

### CODE REVIEW

File: terraform/terraform-module/README.md [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+
+  tags = {
+    Name = var.instance_name
+  }
+}
+```
+
+#### Outputs
+
+Our EC2 instance module exposes the instance ID as an output in `outputs.tf`:
```
Comment (dassiorleando) [inline terraform/terraform-module/README.md:252]: The output is the instance public IP instead, so the description on the top should match.

File: terraform/terraform-module/README.md [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+
+```hcl
+output "instance_ip" {
+  value = aws_instance.instance.public_ip
+}
+```
+
+## Using the Module
```
Comment (dassiorleando) [inline terraform/terraform-module/README.md:260]: Plural form: **Using the Modules**

File: terraform/terraform-module/README.md [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+``` shell
+terraform init
+terraform plan
+terraform apply -auto-approve
+```
+
+Once completed, your infrastructure should resemble the image displayed below:
+
+![Infrastructure](./images/infrastructure.png)
+
```
Comment (dassiorleando) [inline terraform/terraform-module/README.md:316]: Let's add the source code link after the article content:

The complete source code of the project is available on [GitHub](https://github.com/numerica-ideas/community/tree/master/terraform/terraform-module).

Comment (dassiorleando) [inline terraform/terraform-module/README.md:316]: Before the Discord block here, could we add a call-to-action to another Terraform article?

Example: 

In order to better your DevOps skills, learn [How To Deploy WordPress on a 2-Tier AWS Architecture using Terraform](https://blog.numericaideas.com/deploy-wordpress-2-tier-aws-architecture-with-terraform) by following this practical workshop:

[FeaturedImage](https://blog.numericaideas.com/deploy-wordpress-2-tier-aws-architecture-with-terraform)

File: terraform/terraform-module/README.md [hunk 4]
```diff
@@ -0,0 +1,330 @@
+# Demystifying Terraform Modules: A Comprehensive Guide&nbsp;[![Hits](https://hits.seeyoufarm.com/api/count/incr/badge.svg?url=https%3A%2F%2Fgithub.com%2Fnumerica-ideas%2Fcommunity%2Ftree%2Fmaster%2Fterraform%2Fmodule&count_bg=%2379C83D&title_bg=%23555555&icon=&icon_color=%23E7E7E7&title=hits&edge_flat=false)](https://blog.numericaideas.com/terraform-module)
```
Comment (dassiorleando) [inline terraform/terraform-module/README.md:1]: What about keeping the title as simple as: **Demystifying Terraform Modules**

Without the comprehnsive statement!

**Review by dassiorleando — CHANGES_REQUESTED** [review]
@kemanedonfack could you please take a look at the following points before we push the article live?

Great article 👍

**Comment by dassiorleando** [comment]
Also, let's add this before the **Prerequisites** section:

> The **YouTube Channels** in both English (En) and French (Fr) are now accessible, feel free to subscribe by clicking [here](https://www.youtube.com/@numericaideas/channels?sub_confirmation=1).

File: terraform/terraform-module/README.md (hunk 3 above)
Comment (dassiorleando) [reply terraform/terraform-module/README.md:323]: This link is incorrect in the recent changes, it should go to **numerica-ideas** repo, not to your fork.

Comment (dassiorleando) [reply terraform/terraform-module/README.md:323]: You changed the article link for the image, it should match with the article link on the blog ... not the repo code of your fork.

Comment (kemanedonfack) [reply terraform/terraform-module/README.md:323]: I put a relative link

terraform module

File: terraform/terraform-module/README.md [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+terraform init
+terraform plan
+terraform apply -auto-approve
+```
+
+Once completed, your infrastructure should resemble the image displayed below:
+
+![Infrastructure](./images/infrastructure.png)
+
+The complete source code of the project is available on [GitHub](/terraform/terraform-module/).
```
Comment (dassiorleando) [inline terraform/terraform-module/README.md:324]: Let's use an absolute link here for the source code link, that way it'll be easier when migrating content.

File: terraform/terraform-module/README.md [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+
+
+In order to better your DevOps skills, learn How To [Deploy WordPress on a 2-Tier AWS Architecture using Terraform](https://blog.numericaideas.com/deploy-wordpress-2-tier-aws-architecture-with-terraform) by following this practical workshop
+
+![FeaturedImage](../deploy-wordpress-2tier-aws-architecture-with-terraform/images/Deploying-WordPress-on-a-2-Tier-AWS-Architecture-with-Terraform.png)
```
Comment (dassiorleando) [inline terraform/terraform-module/README.md:329]: The targetted link of this image_link should be: https://blog.numericaideas.com/deploy-wordpress-2-tier-aws-architecture-with-terraform

File: terraform/terraform-module/README.md [hunk 7]
```diff
@@ -0,0 +1,344 @@
...
+
+Terraform modules offer several advantages:
+
+- **Modularity**: Modules break down your infrastructure code into manageable components, focusing on specific aspects. This simplifies understanding and maintenance.
+
+- **Reusability**: Modules can be shared across projects, encapsulating best practices for replicating infrastructure patterns.
+
+- **Versioning**: Modules can be versioned, ensuring consistent configurations throughout your infrastructure.
+
+## How to Create Terraform Modules
```
Comment (dassiorleando) [inline terraform/terraform-module/README.md:42]: Sorry, I misguided you here, we should have the question mark at the end.

**How to Create Terraform Modules?**

File: terraform/terraform-module/README.md [hunk 8]
```diff
@@ -0,0 +1,344 @@
...
+
+### Why Use Terraform Modules?
+
+Terraform modules offer several advantages:
+
+- **Modularity**: Modules break down your infrastructure code into manageable components, focusing on specific aspects. This simplifies understanding and maintenance.
+
+- **Reusability**: Modules can be shared across projects, encapsulating best practices for replicating infrastructure patterns.
+
+- **Versioning**: Modules can be versioned, ensuring consistent configurations throughout your infrastructure.
```
Comment (dassiorleando) [inline terraform/terraform-module/README.md:40]: I do believe that we should add a sub-section here that talks about the **Terraform Registry** as a way to reuse other public modules created by the community and that it's a trusted source managed by **Hashicorp**.

Hint: https://blog.numericaideas.com/introduction-to-terraform#provider

File: terraform/terraform-module/README.md [hunk 9]
```diff
@@ -0,0 +1,344 @@
...
+Using a Terraform module in your configuration is straightforward. Specify the module source in your code, like this:
+
+```hcl
+module "example" {
+  source          = "./modules/my-module"
+  instance_count  = 2
+}
+```
+
+In this example, we use a module named **example** located in the `./modules/my-module` directory. We also provide a value for the `instance_count` variable.
```
Comment (dassiorleando) [inline terraform/terraform-module/README.md:94]: In this section, could you give some info about how we could have **reused/installed** that module if it was stored in the **Terraform Registry** or located in a **GitHub repository**?

Simply knowing the possibilities is great, even if the practical example won't cover these cases.

**Review by dassiorleando — CHANGES_REQUESTED** [review]
More review comments are coming your way @kemanedonfack
I hope these will help make this piece of content even better.

File: terraform/terraform-module/README.md [hunk 10]
```diff
@@ -0,0 +1,376 @@
+# Demystifying Terraform Modules&nbsp;[![Hits](https://hits.seeyoufarm.com/api/count/incr/badge.svg?url=https%3A%2F%2Fgithub.com%2Fnumerica-ideas%2Fcommunity%2Ftree%2Fmaster%2Fterraform%2Fmodule&count_bg=%2379C83D&title_bg=%23555555&icon=&icon_color=%23E7E7E7&title=hits&edge_flat=false)](https://blog.numericaideas.com/terraform-module)
+
+**This article was originally written by "Kemane Donfack" on the blog**: https://blog.numericaideas.com/terraform-module
+
+In the world of Infrastructure as Code (IaC), Terraform is a leading tool for managing and provisioning resources across various cloud providers. It empowers DevOps teams to define infrastructure in a declarative manner, ensuring consistent and reproducible deployments. Key to Terraform's capabilities are **modules**, which play a pivotal role in creating scalable and maintainable infrastructure code.
+
```
Comment (dassiorleando) [inline terraform/terraform-module/README.md:6]: Here's the featured image to use for the article, just before the YouTube call-to-action message:

Terraform Modules

Another example: https://github.com/kemanedonfack/community/tree/terraform/terraform/deploy-wordpress-2tier-aws-architecture-with-terraform

Comment (kemanedonfack) [reply terraform/terraform-module/README.md:6]: Nice

File: terraform/terraform-module/README.md (hunk 6 above)
Comment (dassiorleando) [reply terraform/terraform-module/README.md]: > The targetted link of this image_link should be: https://blog.numericaideas.com/deploy-wordpress-2-tier-aws-architecture-with-terraform

[Still not done]

**Pull request merged**